sdk/core/azure-core/inc/http/curl/curl.hpp

```diff
@@ -26,26 +27,24 @@ namespace Azure { namespace Core { namespace Http {
     constexpr static const char* c_DefaultFailedToGetNewConnectionTemplate
         = "Fail to get a new connection for: ";
     constexpr static int c_DefaultMaxOpenNewConnectionIntentsAllowed = 10;
-  } // namespace Details
-
-  /**
-   * @brief Statefull component that controls sending an HTTP Request with libcurl thru the wire and
-   * parsing and building an HTTP RawResponse.
-   * This session supports the classic libcurl easy interface to send and receive bytes from network
-   * using callbacks.
-   * This session also supports working with the custom HTTP protocol option from libcurl to
-   * manually upload and download bytes using a network socket. This implementation is used when
-   * working with streams so customers can lazily pull data from netwok using an stream abstraction.
-   *
-   * @remarks This component is expected to be used by an HTTP Transporter to ensure that
-   * transporter to be re usuable in multiple pipelines while every call to network is unique.
-   */
-  class CurlSession : public BodyStream {
-    // connection handle. It will be taken from a pool
+// Cleaner runs every 5 min or every 1 second for unit tests
+#ifdef TESTING_BUILD
```

Comment:
I don't think changing program behavior via this macro is a good idea.

Reply:
i'd rather we don't have ut and do target test

Reply:
Because intuitively customers won't expect the macro `BUILD_TESTING` would make any changes to product code. It's only supposed to control whether or not build test.

Reply:
Not having the test at all is not better.
The only thing I am changing here are the time intervals which should not make any difference against release build.

The same way I have added some hooks for unit tests to read private state for validation only.

I can move this test to be nightly test and use the same values from Release build. However, we still need the test hooks to read private state.

Reply:
My point is what we're testing being different from what our customers actually see is not a good idea.

Reply:
Agree. That's not ideal.
That's just a reference and starting point.
As soon as we have a nightly test job I remove this workaround and move the test to be run with the same time intervals as how customers will use it.

For now, I just want to make sure that regardless interval time, the cleaning function can do what it have to do.

Reply:
Sure, but can you create an issue tracking that? In case we'd forget.

Reply:
Another question is do we have to cover everything in UT? I think for some functionality targeting test will be enough.

Reply:
issue created: https://github.com/Azure/azure-sdk-for-cpp/issues/533

Reply:
about the question.
IMHO we should have a rule about this and keep code coverage as close to 100% as possible.
We could use mocks or the test recorder for test data. But yes, if you ask me, there should be unit tests for all the code we have.